client/src/components/modals_and_popovers/FocusLockedModal.js

```diff
@@ -0,0 +1,44 @@
+import classNames from "classnames";
+import DOMPurify from "dompurify";
+import React from "react";
+import { Modal } from "react-bootstrap";
+import FocusLock from "react-focus-lock";
+
+import "./bootstrap_modal_exstension.scss";
+
+export class FocusLockedModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    if (!props.ariaLabel) {
+      throw new Error("Must have a prop ariaLabel");
+    }
```

Comment:
Only asserted at construction, what if the prop was updated to be empty later?

If you want to properly mark a prop required (and may as well mark it as a string type while we're at it), look in to [React propTypes](https://reactjs.org/docs/typechecking-with-proptypes.html).